File: home/eax/music/rename.pl
#!/usr/bin/env perl

use strict;
use warnings;

while(my $fname = <*.mp3>) {
  my $at = `mp3info -p "%a - %t.mp3" "$fname"`;
  if($at ne $fname) {
    printf("$fname -> $at\n");
    rename($fname, $at);
  }
}

File: home/eax/data/music/rename.py
#!/usr/bin/env python3

import subprocess
import glob
import os

for fname in glob.glob("*.mp3"):
    cmd = """mp3info -p "%a - %t.mp3" -- "{}" """.format(fname)
    at = subprocess.check_output(cmd, shell=True)
    at = at.decode('ascii', 'replace').strip()
    if fname != at:
        print("{} -> {}".format(fname, at))
        os.rename(fname, at)
